packages/lit-dev-content/site/docs/v3/tools/requirements.md

```diff
@@ -49,3 +49,38 @@ All modern browsers update automatically and users are highly likely to have a r
 ## Note on legacy browsers {#note-on-legacy-browsers}
 
 Lit 3 is not tested on legacy browsers, specifically Internet Explorer 11 and Classic Edge are not supported due to non-standard DOM behavior. If you must support legacy browsers, consider using Lit 2 with additional compilation and/or polyfills as described in [Building for legacy browsers](/docs/v2/tools/requirements#building-for-legacy-browsers).
+
+## Multiple versions of Lit {#multiple-lit-versions}
+
+If Lit is being used as an internal dependency of elements, elements can use different versions of Lit.
+We also take care to ensure that Lit 2 and Lit 3 are mostly interoperable with each other. For example, you can pass a Lit 2 template into a Lit 3 render function and vice-versa.
+
+Because having multiple compatible versions of Lit loaded is non-optimal, due to shipping extra duplicated bytes to the user, we raise a development mode warning.
+
+If you’re publishing a library that uses Lit, follow our [publishing best practices](https://lit.dev/docs/tools/publishing/#don't-bundle-minify-or-optimize-modules) so consumers of your library are able to de-duplicate Lit in their projects.
+
+### Resolving multiple versions of Lit
+
+If you’re seeing a `Multiple versions of Lit loaded` development mode warning, there are a couple things you can try.
+
+1. Find out which Lit libraries have multiple versions loaded by checking the following variables in your browser console: `window.litElementVersions`, `window.reactiveElementVersions`, and `window.litHtmlVersions`.
+
+2. Use `npm ls` (note, you can specify exact libraries to look for, e.g. `npm ls @lit/reactive-element`) to narrow down which dependencies are loading multiple different versions of Lit.
+
+3. Try to use `npm dedupe` to de-duplicate Lit. Use `npm ls` to verify if the duplicated Lit package was successfully de-duped.
+
+4. If there is still duplication, you may need to delete your package lock and `node_modules`. Then install the version of `lit` you want explicitly, followed by your dependencies.
+
+It is possible to follow these steps, and not be able to de-duplicate Lit. For example, a library you depend on may bundle a specific version of Lit in a way that cannot be de-duplicated. In these cases the warning can be ignored.
+
+If the warning is distracting and you want to silence it, add the following code early in your application before Lit has been loaded to initialize the global `litIssuedWarnings` set, and preemptively add the warning to the set.
+
+```html
+<script>
+  window.litIssuedWarnings = new Set();
+  // By adding the Lit warning to this global set, Lit will no longer emit the multiple versions warning.
+  window.litIssuedWarnings.add(
+    'Multiple versions of Lit loaded. Loading multiple versions is not recommended. See https://lit.dev/msg/multiple-versions for more information.'
+  );
+</script>
```

Comment:
This feels very hacky. TIL that we have a way to turn off warnings for `ReactiveElement` https://lit.dev/docs/tools/development/#development-build-runtime-warnings, so perhaps we should consider adding this globally so we can suppress `lit-html` warnings too. That is, if we really deem the 'multiple-versions' warning to be optional.

Reply:
Agree that it feels hacky, however I think it's necessary that we document a way to resolve this warning, especially if someone has investigated the warning and cannot de-duplicate due to application specific reasons (maybe a dependency is bundling Lit).

The reasons I think we need to keep this until we have a better alternative is:

 1. If you cannot resolve warnings, they waste developer time as each developer tries to resolve them. Mentioned [here](https://github.com/lit/lit/pull/4319#issuecomment-1782389845)
 2. If warnings build up because you cannot resolve them, they start to just become background noise that'll disappear in production.
 
This solution is fragile, but it is compatible with Lit 2 and Lit 3, and a very small amount of code.

Reply:
Now that I think of this, I wish we hadn't done the deduping of warnings by the message string. We'd be opting out of the dedupe across versions if we want to change the message.

Still, writing out our "internal" mechanism for deduping warnings just feels wrong. Feels like documenting a way to hack our own package. I'd almost prefer it if users patch `console.warn` to skip any message that includes `'https://lit.dev/msg/multiple-versions'`.

I'm not happy, but I don't wish to block the rest of this excellent addition on account of this.

Reply:
I agree that we probably should not have de-duped based on the string contents, but instead some stable error code that we've invented. E.g. Rust error code index: https://doc.rust-lang.org/error_codes/error-index.html

We should probably think about adopting some unique error string, or maybe we could use the first argument that becomes the link to lit.dev to track duplication.

> I'm not happy, but I don't wish to block the rest of this excellent addition on account of this.

Your message swayed me with the note that we can easily hack around this without relying on Lit globals/internals. Since the `console.warn` hack can easily behave in the same way I agree with you to not document the specific hack.


I removed it. WDYT about adding a note that you can silence it by patching `console.warn`? Or should we leave that to users?

Reply:
Discussed offline–documenting hacks in official docs is probably not a great idea.

Reply:
Leaning towards not mentioning and leaving it up to users. We can revisit if users really want a way to suppress it. And perhaps we can provide a longer term solution similar the reactive-element's disable warning feature though that'll only work for future versions from that point on. we have to start somewhere!